avaje-record-builder/src/main/java/io/avaje/recordbuilder/RecordBuilder.java

```diff
@@ -3,38 +3,76 @@
 import static java.lang.annotation.ElementType.MODULE;
 import static java.lang.annotation.ElementType.PACKAGE;
 import static java.lang.annotation.ElementType.TYPE;
-import static java.lang.annotation.RetentionPolicy.CLASS;
 import static java.lang.annotation.RetentionPolicy.SOURCE;
 
+import java.lang.annotation.Annotation;
 import java.lang.annotation.Documented;
-import java.lang.annotation.ElementType;
 import java.lang.annotation.Retention;
 import java.lang.annotation.Target;
 
-/**
- * An annotation for service providers as described in {@link java.util.ServiceLoader}. The
- * annotation processor generates the configuration files that allow the annotated class to be
- * loaded with {@link java.util.ServiceLoader#load(Class)}.
- *
- * <p>The annotated class must conform to the service provider specification. Specifically, it must:
- *
- * <ul>
- *   <li>be a non-inner, non-anonymous, concrete class
- *   <li>have a publicly accessible no-arg constructor
- * </ul>
- */
+import org.jspecify.annotations.Nullable;
+
+/** Generate a builder class for the given record */
 @Documented
 @Target(TYPE)
 @Retention(SOURCE)
 public @interface RecordBuilder {
 
+  /** Whether getter methods should be generated on the builder */
   boolean getters() default false;
 
+  /**
+   * Whether generated setter methods will enforce null safety for all members (excluding any
+   * annotated with any form of {@code @Nullable}). Has the same effect as annotating each component
+   * with {@code @NonNull}
+   */
+  boolean enforceNullSafety() default false;
+
+  /** What nullable annotation to use for the builder fields. */
+  Class<? extends Annotation> nullableAnnotation() default Nullable.class;
+
+  /** The interfaces the generated builder will extend. */
+  Class<?>[] builderInterfaces() default {};
+
   @Retention(SOURCE)
   @Target({TYPE, PACKAGE, MODULE})
   @interface Import {
 
     /** Specify types to generate Builders for. */
     Class<? extends Record>[] value();
+
+    /** Whether getter methods should be generated on the builder */
+    boolean getters() default false;
+
+    /**
+     * Whether generated setter methods will enforce null safety for all members (excluding any
+     * annotated with any form of {@code @Nullable}). Has the same effect as annotating each
+     * component with {@code @NonNull}
+     */
+    boolean enforceNullSafety() default false;
+
+    /** What nullable annotation to use for the builder fields. */
+    Class<? extends Annotation> nullableAnnotation() default Nullable.class;
+
+    /** The interfaces the generated builder will extend. */
+    Class<?>[] builderInterfaces() default {};
+  }
+
+  @Retention(SOURCE)
+  @Target({TYPE, PACKAGE, MODULE})
+  @interface GlobalConfig {
```

Comment:
In terms of supporting partial compilation this state generally needs to be stored and retrieved. For example, a partial compile rebuilds a builder but does not include the package etc that GlobalConfig is on.

That is, I didn't see any code that stores this state and reloads this state - so I don't think this supports a partial compile that doesn't include the GlobalConfig.